### PR TITLE
Added opt arg for getting coverage from index.html (output by nosetests)

### PR DESCRIPTION
**Pull request by WindfallLabs**
Great little package here! But it was giving me a different coverage % than what the index.html (and other files) said I had and I'd like to use that %.  
I ran nosetests like this: `nosetests --with-coverage --cover-package=<pkg> --cover-html`  

This is my first contribution to another person's work, so thanks for considering my request.

### CODE REVIEW

**Comment by dbrgn** [comment]
Hi, thanks for the contribution!

Is this generated HTML specific to nosetests or is it the same if you just generate a report with `coverage` directly?

**Comment by WindfallLabs** [comment]
The one that nose produced for me is identical (within reason) to the one [here](http://nedbatchelder.com/files/sample_coverage_html/index.html). I tested `get_from_html()` on it and it worked.

**Comment by dbrgn** [comment]
That sounds good, thanks for checking!

But still, before merging this I'd like to know the reason for the difference in the numbers (maybe I have a bug in my code). Do you have additional configuration that might affect the coverage count? The project in question isn't public, is it?

**Comment by WindfallLabs** [comment]
I think that coverage.py considers all modules in a package, but my tests only currently cover code imported by __init__.py causing nose to calculate coverage for just two .py files out of about five. Any additional code imported in __init__ will then be covered by nose as the dev process rolls along. So in the end, I'd like the coverage badge to only reflect implemented code and ignore my work-in-progress code.  

I use a .cfg file that tells nose to use `--with-coverage --cover-erase --cover-package=dslw --cover-html`. And yes, `dslw` is a public repo.  

Maybe there's a better way than the one I proposed to tell your script to only cover the implemented modules of a package, but since nose does this already, I figured I'd make something that works for me/nose. If this is lying about the coverage of a project, maybe this function could add an "*" to indicate that not all code was actually covered.

**Comment by dbrgn** [comment]
Sorry for the long wait and thanks for your explanations. I'm not exactly sure about the relation between nose and coverage, but I'll merge this PR after a few small details are fixed. I'll add comments :)

File: README.rst [hunk 1]
```diff
@@ -40,13 +40,14 @@ It's important that you run ``coverage-badge`` from the directory where the
...
 
-    usage: __main__.py [-h] [-o FILEPATH] [-q] [-v]
+    usage: __main__.py [-h] [-o FILEPATH] [-g HTML] [-q] [-v]
 
     Generate coverage badges for Coverage.py.
 
     optional arguments:
       -h, --help   show this help message and exit
       -o FILEPATH  Save the file to the specified path.
+      -n, --html HTML  Get coverage from index.html output by nosetests.
```
Comment (dbrgn) [inline README.rst]: The help does not match the letter used in the `usage` line.

Could you change the single-letter flag to `-d` (like "directory")? Also, you can remove the `by nosetests` part.

File: coverage_badge/__main__.py [hunk 2]
```diff
@@ -35,6 +36,17 @@ def get_total():
     return '{0:.0f}'.format(total)
 
 
+def get_from_html(index_html):  # new
+    """
+    Return the total from index.html.
+    """
+    with open(index_html, "rb") as f:
```
Comment (dbrgn) [inline coverage_badge/__main__.py]: I'd probably open the file in text mode (so `open(indx_html, 'r') as f`). Or is there a specific reason it's using binary mode?

File: coverage_badge/__main__.py [hunk 3]
```diff
@@ -52,6 +64,8 @@ def parse_args(argv=None):
     parser = argparse.ArgumentParser(description=__doc__)
     parser.add_argument('-o', dest='filepath',
             help='Save the file to the specified path.')
+    parser.add_argument('-n', '--html', dest='html',                    # new
+            help='Get coverage from index.html output by nosetests.')   # new
```
Comment (dbrgn) [inline coverage_badge/__main__.py]: You can remove the `# new` comments :)

**Comment by pmourelle** [comment]
I'd love to see this merged. Being a while though. @WindfallLabs are you continuing this?

File: coverage_badge/__main__.py (hunk 3 above)
Comment (WindfallLabs) [inline coverage_badge/__main__.py]: Sorry I haven't played with this much in a while. The project that was going to use this hasn't gotten enough attention lately and this is my first contribution to a project. Not off to a great start...  
No reason for binary mode to answer that question from way back, and I have no preferences on the -n vs -d options. Glad there's interest @pmourelle!

File: README.rst [hunk 4]
```diff
@@ -40,13 +40,14 @@ It's important that you run ``coverage-badge`` from the directory where the
 
 The full usage text::
 
-    usage: __main__.py [-h] [-o FILEPATH] [-q] [-v]
+    usage: __main__.py [-h] [-o FILEPATH] [-g HTML] [-q] [-v]
```
Comment (dbrgn) [inline README.rst]: Another typo here, this should be `-d`, right?

File: README.rst (hunk 1 above)
Comment (dbrgn) [inline README.rst]: Again, here the change to `-d` is missing.

File: README.rst (hunk 4 above)
Comment (WindfallLabs) [inline README.rst]: So we're changing -g and -n to -d?

Comment (dbrgn) [inline README.rst]: Well, it needs to be consistent, and the argument is defined as `-n` (see https://github.com/dbrgn/coverage-badge/pull/1/files#diff-59ad6794e1dd5e79d9c48b11b7953128R67), so the documentation should say the same :)

**Pull request closed without merging**